### ISSUE TITLE
IPython Website Revamp

### ISSUE THREAD

**Issue by cameronoelsen** [jupyter/design#15]
Over the last week, I have been doing a visual overhaul for both IPython and Jupyter. On the site I am sharing below, I simply took the content from the IPython site and did visual changes to make the site more modern. However, some of the content on the mockup provided is going to be moved over to the Jupyter site. The IPython site will be most likely less content heavy and more-so a direct to the Jupyter site once it is made. I have also allowed commenting on the mockup, so please feel free to provide feedback if you have any! The mockup is also very functional, so play around :) Cheers!

http://invis.io/KF3BRI8QU

**Comment by rgbkrk** [comment]
That's pretty great. Had to stand like 5 feet from my monitor though and invis.io didn't let me zoom out.

**Comment by Carreau** [comment]
A few comment inline. Looks nice !

General questions. Do we want to get closer / farter to Python Blue? And Yellow?

**Comment by ellisonbg** [comment]
Kyle, do you mean that the font and design is too big?

On Mon, Jun 22, 2015 at 11:16 PM, Matthias Bussonnier <
notifications@github.com> wrote:

> A few comment inline. Looks nice !
>
> General questions. Do we want to get closer / farter to Python Blue? And
> Yellow?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/design/issues/15#issuecomment-114374888>.
>



-- 
Brian E. Granger
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by rgbkrk** [comment]
Yes. I posted that way too late and it came out really badly. Sorry @cameronoelsen.

I posted in the invis.io as well.

**Comment by Carreau** [comment]
BTW, I suppose the @ellisonbg and @cameronoelsen have seen this: http://arstechnica.com/gadgets/2015/06/photoshop-gets-an-open-source-html5-based-interface-for-app-design/

**Comment by damianavila** [comment]
That's great, I agree with @rgbkrk about the font size... but at first sight, I like it...

**Comment by neilpanchal** [comment]
Looks great! I just have one criticism: Too many "flash-cards" to scroll through. I personally prefer the current style with better fonts and minimal design. Basically, the functional objective of a site like iPython/Jupyter is to get the core information relayed to the viewer as fast as possible. I kind of like www.julialang.org honestly. It is boring and that's precisely why I like it. Pff....I sound like an old grumpy man :+1:

**Comment by jdfreder** [comment]
I think this design is at least one to two million times better than our current website design.  However, I think @rgbkrk is right, it feels a like bloated on a large monitor (I'm not sure if that's the font-sizes or what).

**Comment by Carreau** [comment]
Well I think we need both kind of design, Julia lang is imho, too minimal, and can be scary to the novice user. Though it has all the information density that programers search to have, numbers, code example. Python.org is terrible at that.

Seeing how bad our site is, I think anyway we need many things:
  - a Rough redesing is something we like.
  - re write the contents.
  - finish design. 

Because right now the content is driving the design and I can perfectly see some huge changes like actually have a thebe cell on the main page, as in the end IPython will be "just" a kernel for Jupyter.

**Comment by jdfreder** [comment]
> Though it has all the information density that programers search to have, numbers, code example.

I think that kind of information density definitely does not belong on the landing page.

Why not design first, then update the content, and then iterate on the design some more?

**Comment by Carreau** [comment]
Fair enough, I'll try to review and be more critic on the design. I don't know how @cameronoelsen works. 
Do you prefer to get pixel perfect mock-up ? Or is it a "rough idea" we iterate over an actual implementation ?

**Comment by Carreau** [comment]
(Also I guess you have face to face meeting with @ellisonbg, when this happen you can also post here a quick recap of what has been said, otherwise you will get contradicting things to do from both sides)

**Comment by ellisonbg** [comment]
It will be much easier/faster to iterate on the visual style before we jump
to implementation.

If I have in person meetings, I will record notes of the feedback here.

But we will have to start iterating on content before we can really finish
the visual design. My plan is to create a hackpad we can use to outline the
different pages and content we want on the two websites.

On Wed, Jun 24, 2015 at 11:59 AM, Matthias Bussonnier <
notifications@github.com> wrote:

> Fair enough, I'll try to review and be more critic on the design. I don't
> know how @cameronoelsen <https://github.com/cameronoelsen> works.
> Do you prefer to get pixel perfect mock-up ? Or is it a "rough idea" we
> iterate over an actual implementation ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/design/issues/15#issuecomment-114980584>.
>



-- 
Brian E. Granger
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by cameronoelsen** [comment]
I am starting to look through both sites to see what we want for each page, but I can't really design anything until I know for sure what content will be on each page and how long the sections will be. This makes it easier for me to get a sense of what type of information it is and how to design it accordingly. Currently, I have designed the top of the front page and found that most of the content on the past IPy mockup is pretty pertinent to the Jupyter site (Announcements, About, Citing etc.). I have included below a mockup of the top of the screen. Below this top section follows About, Announcements, Cite and other important information about Jupyter. Instead of the blue color scheme, it is going to follow the orange that the Jupyter logo has inside of the main "planet". Lastly, I have attached a hackpad to collaborate on both what each page on the site should be called as well as what content each page should hold (each section with what description/content inside). Thanks and I look forward to further design the site!! :)
![jupyter home1](https://cloud.githubusercontent.com/assets/6437976/8343140/0fbf77e0-1a87-11e5-8357-a5ee4b1d3663.png)
https://jupyter.hackpad.com/Jupyter-Content-and-Organization-Gg50rKW9YW7

**Comment by rgbkrk** [comment]
WHOA! I love that mockup!

**Comment by jhamrick** [comment]
Me too!

**Comment by rgbkrk** [comment]
Since it's an intro page we should probably say "X languages and more to come". People won't necessarily know what a kernel is until working with the project for a bit.

**Comment by damianavila** [comment]
> Since it's an intro page we should probably say "X languages and more to come". People won't necessarily know what a kernel is until working with the project for a bit.

Sound reasonable...

**Comment by jdfreder** [comment]
beautiful!

**Comment by ellisonbg** [comment]
Really like where this is going! More specific feedback tomorrow.

On Wed, Jun 24, 2015 at 5:36 PM, Jonathan Frederic <notifications@github.com
> wrote:

> beautiful!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/design/issues/15#issuecomment-115055013>.
>



-- 
Brian E. Granger
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by rgbkrk** [comment]
> More specific feedback tomorrow.

I'll be more specific now. The grayed out many languages to the left and right is a really good collage. It brings Jupyter to the center while showing the interoperability.

**Comment by fperez** [comment]
Love it! Would it be too much work for the language images to have on-hover events, where on hover they go from grayed-out to full-color and they actually link to their kernel page?

Question: is it meant to not scroll down at all? My impression is that, in this day and age of mobile, we should put some more content at the bottom to be loaded on scroll...

But I do love where this is going, thanks so much.

**Comment by fperez** [comment]
BTW, I think we should split this into an issue for the IPython one and one for the Jupyter one. Else the conversation is going to get very tricky to follow...

**Comment by cameronoelsen** [comment]
@fperez The hover idea sounds great! If we can implement that, would be awesome. Also, it is meant to scroll, I just haven't designed it yet. Will have further designs when the content of the website is sorted out. Also, great idea about two different issues. I will start a new one now.

**Comment by Carreau** [comment]
the hovering icons-coloring might be painful to do if it is not a square map and in particular difficult if we need to update. Also why some languages and not others ? I would prefer a page like @bollwyvl's one with a design that matches.

**Comment by Carreau** [comment]
Moved my message to relevant issues about Jupyter Site

**Comment by fperez** [comment]
Should we be taking periodic screenshots of the design and pasting them here, to make it a bit easier to follow this thread later on? Otherwise, whenever we refresh the URL, we're just going to see whatever the latest version is...

**Comment by cameronoelsen** [comment]
Here is a mockup of what the site will look like on mobile, it is best to look at it on your smartphone! Again, the content is not set in stone and this is just to show how it will scale. More to come! 
http://invis.io/2J3EKEAD8

**Comment by cameronoelsen** [comment]
@fperez Yeah, here are all the current designs (1st iteration) for desktop viewing:
![home - desktop](https://cloud.githubusercontent.com/assets/6437976/8387638/1c116704-1c0f-11e5-9fda-2162d0a7b518.png)
![install - desktop](https://cloud.githubusercontent.com/assets/6437976/8387639/1ed2183a-1c0f-11e5-9e82-0315f5a108aa.png)
![docs - desktop](https://cloud.githubusercontent.com/assets/6437976/8387653/31287df8-1c0f-11e5-9d77-2ed71172dc5a.png)
![videos - desktop](https://cloud.githubusercontent.com/assets/6437976/8387655/34d50520-1c0f-11e5-9933-41e9135a5551.png)
![news - desktop](https://cloud.githubusercontent.com/assets/6437976/8387656/37131606-1c0f-11e5-9266-2bcc1ffe307b.png)
![cite - desktop](https://cloud.githubusercontent.com/assets/6437976/8387659/3b6b3936-1c0f-11e5-9fb1-826665e22f0a.png)
![sponsor - desktop](https://cloud.githubusercontent.com/assets/6437976/8387661/4042c550-1c0f-11e5-90fe-6c6a4f2a1e97.png)
![donate - desktop](https://cloud.githubusercontent.com/assets/6437976/8387662/430d2e60-1c0f-11e5-8fc6-0a279fc99046.png)

**Comment by cameronoelsen** [comment]
And here are all of the mockups (1st iteration) of the IPython site for mobile viewing:
![smartphone home](https://cloud.githubusercontent.com/assets/6437976/8387678/6fed9352-1c0f-11e5-82ed-4ac612416f5e.png)
![install smartphone](https://cloud.githubusercontent.com/assets/6437976/8387681/767f3a9a-1c0f-11e5-85ae-02b031c02197.png)
![documentation smartphone](https://cloud.githubusercontent.com/assets/6437976/8387685/7b0cd806-1c0f-11e5-8d1e-cf0213e2985f.png)
![videos 1 - smartphone](https://cloud.githubusercontent.com/assets/6437976/8387688/7efaf3b2-1c0f-11e5-91d4-298f080435a3.png)
![news - smartphone](https://cloud.githubusercontent.com/assets/6437976/8387691/85d9ab88-1c0f-11e5-8e92-b52232f68e35.png)
![cite - smartphone](https://cloud.githubusercontent.com/assets/6437976/8387695/8b681418-1c0f-11e5-9b57-88d826671229.png)
![sponsor - smartphone](https://cloud.githubusercontent.com/assets/6437976/8387697/8ec9a1b2-1c0f-11e5-8300-894aaf9f131d.png)
![donate - smartphone](https://cloud.githubusercontent.com/assets/6437976/8387698/92068f98-1c0f-11e5-936d-8cc0218c8ef2.png)

**Comment by cameronoelsen** [comment]
@fperez I am keeping folders on the IPython Dropbox with iteration titles to track progress we make, after posting all of the screenshots on here, it made the thread too unreadable and hard to navigate

**Comment by fperez** [comment]
Ah, ok, no worries then. I have them in my email, and indeed, it got huge :)

**Comment by fperez** [comment]
Probably worth taking an occasional screenshot and posting only if there's a specific discussion at one point that goes on about one version, but no need to have the thread have too many...

In any case, great job, I love where this is going.  I've posted a few comments directly on that page since it's so easy to comment on specific visuals there. But we probably want to keep the overall conversation here on the big picture, for future reference.

Thanks for the great work!
